### PR TITLE
Add multithreading support in PML UCX framework.

### PR DESCRIPTION
**Pull request by xinzhao3**

### CODE REVIEW

File: ompi/mca/pml/ucx/pml_ucx.c [hunk 1]
```diff
@@ -178,20 +182,42 @@ int mca_pml_ucx_init(void)
...
     status = ucp_worker_create(ompi_pml_ucx.ucp_context, &params,
                                &ompi_pml_ucx.ucp_worker);
     if (UCS_OK != status) {
         return OMPI_ERROR;
     }
 
+    attr.field_mask = UCP_WORKER_ATTR_FIELD_THREAD_MODE;
+    status = ucp_worker_query(ompi_pml_ucx.ucp_worker, &attr);
+    if (UCS_OK != status) {
+        ucp_worker_destroy(ompi_pml_ucx.ucp_worker);
```
Comment (yosefe) [inline ompi/mca/pml/ucx/pml_ucx.c]: pls add PML_UCX_ERROR message

File: ompi/mca/pml/ucx/pml_ucx.c [hunk 2]
```diff
@@ -178,20 +182,42 @@ int mca_pml_ucx_init(void)
...
+        ompi_pml_ucx.ucp_worker = NULL;
+        return OMPI_ERROR;
+    }
+
+    if (ompi_mpi_thread_multiple && attr.thread_mode != UCS_THREAD_MODE_MULTI) {
+        /* UCX does not support multithreading, disqualify current PML for now */
+        /* TODO: we should let OMPI to fallback to THREAD_SINGLE mode */
+        ucp_worker_destroy(ompi_pml_ucx.ucp_worker);
```
Comment (yosefe) [inline ompi/mca/pml/ucx/pml_ucx.c]: pls add PML_UCX_ERROR message

**Comment by hppritcha** [comment]
@yosefe please check off on the review if its ready to go.

**Comment by hppritcha** [comment]
@bwbarrett good to go!

**Pull request merged**